File: 204. Count Primes/Solution.py
class Solution:
    # @param {integer} n
    # @return {integer}
    def countPrimes(self, n):
        if n <= 1:
            return 0
            
        isPrime = [True] * n
        isPrime[0] = isPrime[1] = False
        
        i = 2
        while i * i < n:
            tmp = i * i
            if isPrime[tmp]:
                while tmp < n:
                    isPrime[tmp] = False
                    tmp += i
            i += 1
            
        return sum(isPrime)

File: 204. Count Primes/Solution.cpp
class Solution {
public:
    int countPrimes(int n) {
        if (n <= 1)
            return 0;
            
        bool isPrime[n];
        isPrime[0] = isPrime[1] = false;
        for (int i = 2; i < n; i++)
            isPrime[i] = true;
        
        for (int i = 2; i * i < n; i++) {
            int tmp = i * i;
            if (isPrime[tmp]) {
                while (tmp < n) {
                    isPrime[tmp] = false;
                    tmp += i;
                }
            }
        }
        
        int ans = 0;
        for (int i = 2; i < n; i++)
            if (isPrime[i])
                ans++;
                
        return ans;
        
    }
};